packages/fetchai/skills/tac_control_contract/behaviours.py

```diff
@@ -149,16 +146,14 @@ def act(self) -> None:
                 self.context.logger.info("Mint objects after registration.")
                 for agent in self.context.configuration.agent_addr_to_name.keys():
                     self._mint_objects(
-                        is_batch=True,
-                        address=agent,
-                        nonce_index=self.context.shared_state["agent_counter"],
+                        is_batch=True, address=agent, nonce_index=self.agent_counter,
```

Comment:
why do we not handle the nonce on the contract.py. Seems like this is unnecessary complexity to expose here?

Reply:
No we handle it in the contract. Though, because we are creating the tx before we send it the transaction, due to the loop need to add the counter so I manually update the nonce here.